### PR TITLE
core: add capability to output message queue

### PR DESCRIPTION
**Pull request by authmillenon**

### CODE REVIEW

File: core/thread_print_msg_queue.c [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file
```
Comment (kaspar030) [inline core/thread_print_msg_queue.c]: Author missing

Comment (authmillenon) [inline core/thread_print_msg_queue.c]: Addressed.

**Comment by authmillenon** [comment]
Rebased to current master.

**Comment by authmillenon** [comment]
(and made some adaptation)

File: core/thread_print_msg_queue.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+void thread_print_msg_queue(void)
+{
+    volatile tcb_t *tcb = sched_active_thread;
+    volatile cib_t *msg_queue = &tcb->msg_queue;
+    msg_t *msg_array = tcb->msg_array;
+    unsigned int i = msg_queue->read_count & msg_queue->mask;
+
+    printf("Message queue of thread %" PRIkernel_pid "\n", tcb->pid);
+    printf("    size: %u (avail: %d)\n", msg_queue->mask + 1,
```
Comment (BytesGalore) [inline core/thread_print_msg_queue.c]: maybe a matter of taste, but I would prefer parens :)

**Comment by BytesGalore** [comment]
Looks good so far I can see the q size of the thread though I don't know how to enforce to print a filled q

**Comment by authmillenon** [comment]
Just put it somewhere where you fill the queue (e.g. before `msg_send()`)

**Comment by BytesGalore** [comment]
ahh thx for the hint.
Tested with `tests/thread_msg_avail` and it works as expected :+1:
Beside the my comment on the parens ACK.

File: core/thread_print_msg_queue.c (hunk 2 above)
Comment (authmillenon) [inline core/thread_print_msg_queue.c]: I don't ^^

**Comment by thomaseichinger** [comment]
Tested and it works. I have no strong opinion on the parentheses, @authmillenon please comment if or if you do not change this. ACK and merge at will.

**Pull request merged**